test/cpp_extensions/open_registration_extension/setup.py

```diff
@@ -40,6 +41,9 @@ def run(self):
             CXX_FLAGS = ["/sdl"]
         else:
             CXX_FLAGS = ["/sdl", "/permissive-"]
+    elif platform.machine() == "s390x":
+        # no -Werror on s390x due to newer compiler
+        CXX_FLAGS = {"cxx": ["-g", "-Wall"]}
```

Comment:
Can we identify the problematic C++ source code lines and suppress warnings there?